src/main/java/com/sparta/popupstore/config/AuthUserResolver.java

```diff
@@ -36,11 +37,16 @@ public Object resolveArgument(
     ) {
         HttpServletRequest request = webRequest.getNativeRequest(HttpServletRequest.class);
         if(request == null) {
-            throw new RuntimeException("No request found");
+            throw new IllegalArgumentException("No request found");
         }
 
         Claims userInfo = jwtUtil.getUserInfoFromRequest(request);
-        return userRepository.findByEmail(userInfo.getSubject())
+        User user = userRepository.findByEmail(userInfo.getSubject())
                 .orElseThrow(() -> new RuntimeException("User not found"));
+
+        if(request.getRequestURI().startsWith("/promotionalEvents/admin") && !UserRole.ADMIN.equals(user.getUserRole())){
+            throw new IllegalArgumentException("User has admin role");
```

Comment:
```suggestion
            throw new IllegalArgumentException("User doesn't have admin role");
```
메시지가 반대로 되어 있는 거 같습니다.